lib/components/user/monitored-trip/trip-basics-pane.js

```diff
@@ -51,26 +51,57 @@ const allDays = [
  * and lets the user edit the trip name and day.
  */
 class TripBasicsPane extends Component {
+  /**
+   * Link to the DOM for the trip name label at the top of the form,
+   * so we can scroll to show the trip name and days if the user attempts
+   * to submit the form and there is a validation error for these data.
+   */
+  tripNameLabelRef = createRef()
+
+  /**
+   * Scroll to the trip name/days fields if submitting and there is an error on these fields.
+   * (inspired by https://gist.github.com/dphrag/4db3b453e02567a0bb52592679554a5b)
+   */
+  _scrollToTripDataIfInvalid = () => {
+    const { isSubmitting, isValidating, errors } = this.props
+    const keys = Object.keys(errors)
+
+    if (keys.length > 0 && isSubmitting && !isValidating) {
+      const { current } = this.tripNameLabelRef
+      if (current) {
+        current.scrollIntoView({ behavior: 'smooth', block: 'start' })
+      }
+    }
+  }
+
+  /**
+   * For new trips only, update the Formik state to
+   * uncheck days for which the itinerary is not available.
+   */
+  _updateNewTripItineraryExistence = prevProps => {
+    const { isCreating, itineraryExistence, setFieldValue } = this.props
+
+    if (isCreating &&
+        itineraryExistence &&
+        itineraryExistence !== prevProps.itineraryExistence
+    ) {
+      allDays.forEach(({ name }) => {
+        if (!itineraryExistence[name].valid) {
+          setFieldValue(name, false)
+        }
+      })
+    }
+  }
+
   componentDidMount () {
     // Check itinerary availability (existence) for all days.
     const { checkItineraryExistence, values: monitoredTrip } = this.props
     checkItineraryExistence(monitoredTrip)
   }
 
   componentDidUpdate (prevProps) {
-    const { isCreating, itineraryExistence, setFieldValue } = this.props
-
-    if (itineraryExistence !== prevProps.itineraryExistence) {
-      // For new trips only,
-      // update the Formik state to uncheck days for which the itinerary is not available.
-      if (isCreating && itineraryExistence) {
-        allDays.forEach(({ name }) => {
-          if (!itineraryExistence[name].valid) {
-            setFieldValue(name, false)
-          }
-        })
-      }
-    }
+    this._updateNewTripItineraryExistence(prevProps)
+    this._scrollToTripDataIfInvalid()
```

Comment:
I somewhat expected formik to have something to handle this. I suppose not, eh? Also, similar to how we highlight the stop viewer row on scroll, it might be good to highlight/fade the invalid items too (probably in the alert-danger bg color).

Reply:
The scrolling is not built-in, so there are code snippets and even a library. Library is cleaner. https://github.com/opentripplanner/otp-react-redux/pull/323/commits/4801a00160dbc1d75bb375d39d2cbb0bb4b4c34e 
Errors were already highlighted in standard Bootstrap red.